src/new/wallet.js

```diff
@@ -925,42 +978,96 @@ class HathorWallet extends EventEmitter {
   }
 
   /**
-   * Create a new token for this wallet
+   * Create SendTransaction object and run from mining
+   * Returns a promise that resolves when the send succeeds
+   *
+   * @param {Transaction} transaction Transaction object to be mined and pushed to the network
+   *
+   * @return {Promise} Promise that resolves with transaction object if succeeds
+   * or with error message if it fails
+   *
+   * @memberof HathorWallet
+   * @inner
+   */
+  handleSendPreparedTransaction(transaction) {
+    const sendTransaction = new SendTransaction({ transaction });
+    const promise = new Promise((resolve, reject) => {
+      sendTransaction.on('send-tx-success', (transaction) => {
+        resolve(transaction);
+      });
+
+      sendTransaction.on('send-error', (err) => {
+        reject(err);
+      });
+    });
+
+    sendTransaction.runFromMining();
```

Comment:
Same here about returning a promise.

Reply:
Done here https://github.com/HathorNetwork/hathor-wallet-lib/pull/221/commits/43cdd56eebf3e443132890aaddd495d24aaf24f5

Reply:
That's great! Thanks!